### PR TITLE
chore: useENSDomains

### PR DESCRIPTION
**Pull request by guanbinrui**
image

## New Hooks

+ `useCurrentVisitingIdentity` the current viewing identity on the profile page.
+ `useENSDomains` Resolve a Twitter handle to many Ethereum addresses (Queries ENS under the hook).

closes #

### CODE REVIEW

File: packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx [hunk 1]
```diff
@@ -24,7 +26,35 @@ const sns: Plugin.SNSAdaptor.Definition = {
     },
     CompositionDialogMetadataBadgeRender: new Map([[PLUGIN_META_KEY, () => PLUGIN_NAME]]),
     GlobalInjection() {
-        return <DonateDialog />
+        const lastRecognizedIdentity = useLastRecognizedIdentity()
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Gitcoin/SNSAdaptor/index.tsx:29]: TODO: Need to delete it before merging this PR.

File: packages/maskbook/src/plugins/Wallet/services/domain.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        case DomainType.ENS:
+            return (await fetchDomainsByTwitterIdCached(twitterId)).map((x) => ({
+                label: '',
+                ownerAddress: x.owner,
+                resolvedAddress: x.owner,
+            }))
+        case DomainType.UNS:
+            throw new Error('To be implemennted.')
+        case DomainType.DNS:
+            throw new Error('To be implemennted.')
```
Comment (UncleBill) [inline packages/maskbook/src/plugins/Wallet/services/domain.ts:18]:
```suggestion
            throw new Error('To be implemented.')
        case DomainType.DNS:
            throw new Error('To be implemented.')
```

File: packages/maskbook/src/plugins/Wallet/apis/ens.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import stringify from 'json-stable-stringify'
+
+async function fetchFromENSTextResolverSubgraph<T>(query: string) {
+    const subgraphURL = 'https://api.thegraph.com/subgraphs/name/dimensiondev/ens-text-resolver-subgraph'
+    if (!subgraphURL) return null
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/Wallet/apis/ens.ts:5]: Remove this line?

**Review by NikoManiac — DISMISSED** [review]

File: packages/dashboard/src/web3/context.ts [hunk 4]
```diff
@@ -70,6 +70,7 @@ export const Web3Context: Web3ProviderType = {
     ),
     getAssetsList: PluginServices.Wallet.getAssetsList,
     getAssetsListNFT: PluginServices.Wallet.getAssetsListNFT,
+    getDomainsList: PluginServices.Wallet.getDomainsByTwitterId,
```
Comment (Jack-Works) [inline packages/dashboard/src/web3/context.ts:73]: `getENSDomains`?

Comment (guanbinrui) [reply packages/dashboard/src/web3/context.ts:73]: To keep it the same style as other RPC methods.

**Review by zhouhanseng — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**